torch/sparse/__init__.py

```diff
@@ -485,3 +486,116 @@ def test_mth(*args, **kwargs):
                 return mth(*args, **kwargs)
 
         return test_mth
+
+
+class sparse_semantics:
+    """A tool to control sparse semantics for autograd.
+
+By default, a sparse tensor is considered as a memory optimization of
+a tensor where the unspecified sparse tensor elements have zero
+value. This assumption defines the so-called sparse semantics.
+
+Alternatively, there exists applications where the unspecified sparse
+tensor elements represents a structural lack of information and there
+is no value assigned to such elements. All the specified sparse tensor
+elements (in addition to carrying a element value), define a masked-in
+predicate that enables computations with such elements. The
+unspecified sparse tensor elements define a masked-out predicate that
+disables the corresponding computations. This assumption defines the
+so-called masked semantics.
+
+The following options exists to manage sparse semantics for autograd:
+
+1. Using a context manager:
+
+   .. code:: python
+
+       run_my_model()  # assuming the default sparse semantics
+
+       with torch.sparse.sparse_semantics(False):
+           run_my_model()  # assuming masked semantics
+
+2. Using a procedural approach:
+
+   .. code:: python
+
+       prev_sparse_semantics = torch.sparse.sparse_semantics.is_enabled()
+       torch.sparse.sparse_semantics.enable()
+
+       run_my_model()  # assuming sparse semantics
+
+       if not prev_sparse_semantics:
+           torch.sparse.sparse_semantics.disable()
+
+3. Using function decoration:
+
+   .. code:: python
+
+       @torch.sparse.sparse_semantics(False)
+       def run_my_model():
+           ...
+
+       run_my_model()  # assuming masked semantics
+
+    """
+    @staticmethod
+    def is_enabled():
+        r"""Returns True if the sparse semantics is enabled.
+
+.. note::
+
+    Use :func:`torch.sparse.sparse_semantics.enable` or
+    :func:`torch.sparse.sparse_semantics.disable` to
+    manage the state of the sparse semantics.
+        """
+        return torch._C._sparse_semantics()
+
+    @staticmethod
+    def enable():
+        r"""Enable sparse semantics for autograd.
+
+.. note::
+
+    By default, the sparse semantics is enabled. Use
+    :func:`torch.sparse.sparse_semantics.is_enabled` to retrieve the
+    current state of sparse semantics.
+
+.. note::
+
+    The sparse semantics flag is effective in both Python and ATen.
+
+    The flag can be locally overridden by the
+    :func:`torch.sparse.sparse_semantics` context manager.
+        """
+        torch._C._set_sparse_semantics(True)
+
+    @staticmethod
+    def disable():
+        r"""Disable sparse tensor invariants checking in sparse tensor constructors.
+
+See :func:`torch.sparse.sparse_semantics.enable` for more information.
+        """
+        torch._C._set_sparse_semantics(False)
+
+    # context manager support
+    def __init__(self, enable=True):
+        self.state = enable
+        self.saved_state = None
```

Comment:
```suggestion
        self.saved_state : Optional[bool] = None
```
to fix the lintrunner error.

Reply:
Applied, thanks!